app/views/DocumentEditView/DocumentEditView.jsx

```diff
@@ -28,28 +28,27 @@ class DocumentEditView extends React.Component {
   }
 
   componentDidMount() {
-    const {actions, route} = this.props
-    const {documentId} = route.params
+    const {actions, contentKey, documentId} = this.props
 
     if (!documentId) {
       actions.startDocument({
-        contentKey: this.contentKey
+        contentKey
       })
     }
 
     window.addEventListener('beforeunload', this.userClosingBrowser)
   }
 
   componentDidUpdate(oldProps) {
-    const {actions, route, state} = this.props
+    const {actions, collection, contentKey, documentId, state} = this.props
     const {state: oldState} = oldProps
-    const {documentId} = route.params
-    const document = state.document[this.contentKey] || {}
-    const oldDocument = oldState.document[this.contentKey] || {}
+    const document = state.document[contentKey] || {}
+    const oldDocument = oldState.document[contentKey] || {}
     const isSaving = (oldDocument.saveAttempts || 0) < document.saveAttempts
 
     // Are there unsaved changes?
     if (
+      !document.isLoading &&
```

Comment:
This fixes a minor bug: if there previously were unsaved changes, but now there are not, then in the short amount of time before the document is loaded (with `wasLoadedFromLocalStorage` being false), the previous version of the document (with `wasLoadedFromLocalStorage` true) is grabbed from Redux, which causes the "unsaved changes" notification to fire.